framework/src/parser/CommandLine.C

```diff
@@ -482,14 +482,8 @@ CommandLine::formatEntry(const CommandLine::Entry & entry) const
   oss << entry.name;
   if (entry.value)
   {
-    oss << "=";
-    const auto & value = *entry.value;
-    const auto has_space = value.find_first_not_of(" ") == std::string::npos;
-    if (has_space)
-      oss << "\"";
-    oss << value;
-    if (has_space)
-      oss << "\"";
+    const auto q = (*entry.value).find(" ") != std::string::npos ? "'" : "";
```

Comment:
let's add a comment

so it's better to not have the \ by the spaces?

Reply:
> so it's better to not have the \ by the spaces?

I think the previous implementation was just never doing what we thought it was?
It was checking if the entry value was _all_ spaces (i.e. only spaces) and double quoting that value.
We are now checking if there's at least one space in the entry value and single quoting the value.
The single quotes are neater and avoid the need to escape the character with a backslash.

Reply:
> It was checking if the entry value was all spaces (i.e. only spaces)

I dont think so. It should have been finding if there were any spaces in the string. same as now
find_first_not_of is
```
  // Finds the first occurrence of any of the characters not in `s` within the
  // `string_view`, returning the start position of the first non-match, or
  // `npos` if no non-match was found.
  size_type find_first_not_of(string_view s, size_type pos = 0) const noexcept;
```

> The single quotes are neater and avoid the need to escape the character with a backslash.

works for me

Reply:
> I dont think so.

https://godbolt.org/z/roov4vTWx